app/main.py

```diff
@@ -1,2 +1,334 @@
-class Dictionary:
-    pass
+from __future__ import annotations
+from typing import TypeVar, Any, Self, Mapping
+from dataclasses import dataclass, field
+
+
+_TK = TypeVar("_TK")
+_TV = TypeVar("_TV")
+
+
+@dataclass
+class Node:
+    key: _TK
+    hash_key: int = field(init=False, repr=False)
+    value: _TV
+
+    def __post_init__(self) -> None:
+        self.hash_key = hash(self.key)
+
+
+class DictionaryKeys[_TK]:
+
+    def __init__(self, dictionary: Dictionary[_TK, _TV], /) -> None:
+        self._keys = tuple(
+            node.key for node in dictionary.__dict__["_hash_table"]
+            if node is not None
+        )
+
+    def __iter__(self) -> DictionaryKeysIterator[_TK]:
+        return DictionaryKeysIterator(self)
+
+
+class DictionaryValues[_TV]:
+
+    def __init__(self, dictionary: Dictionary[_TK, _TV], /) -> None:
+        self._values = tuple(
+            node.value for node in dictionary.__dict__["_hash_table"]
+            if node is not None
+        )
+
+    def __iter__(self) -> DictionaryValuesIterator[_TV]:
+        return DictionaryValuesIterator(self)
+
+
+class DictionaryItems[_TK, _TV]:
+
+    def __init__(self, dictionary: Dictionary[_TK, _TV], /) -> None:
+        self._items = tuple(
+            node.value for node in dictionary.__dict__["_hash_table"]
```

Comment:
The `DictionaryItems` class should store key-value pairs, but it currently only stores values. Modify the implementation to store tuples of key-value pairs by changing `node.value` to `(node.key, node.value)`.